mountpoint-s3/src/metrics.rs

```diff
@@ -61,6 +67,16 @@ pub fn install() -> MetricsSinkHandle {
     handle
 }
 
+/// Report process level metrics
+fn poll_process_metrics(sys: &mut System) {
+    if let Ok(pid) = get_current_pid() {
+        sys.refresh_process_specifics(pid, ProcessRefreshKind::new().with_memory());
+        if let Some(process) = sys.process(pid) {
+            metrics::gauge!("mountpoint_s3.total_memory_usage").set(process.memory() as f64);
```

Comment:
nit: something like `process.memory_usage`

also, not sure how useful this is without also knowing how much memory the system has, is that something we can easily get?

Reply:
+1, I've contemplated for a while about this method:
https://docs.rs/sysinfo/latest/sysinfo/struct.System.html#method.available_memory

it likely won't be accurate inside a container (without additional effort from a user), but still seems useful

Reply:
yeah, will add this